### ISSUE TITLE
rfc8: add description and transitions of program states

### ISSUE THREAD

**Issue by chu11** [flux-framework/rfc#62]
I can add later once I know more about flux, but would be good to add short description of states and also the potential transitions between them.

**Comment by dongahn** [comment]
@grondo can chime in. Given that wreck will be replaced, are we ready to lock in those states and state transitions? I have some states from the schedulers that I can add and describe although I don't recall if we wanted to cover the state changes needed by the scheduler. One big unknown in my mind is exact states and state transitions needed for dynamic scheduling.

**Comment by grondo** [comment]
I don't think we're ready to lock in the states at all, but we could certainly at least add description to some of the terms we've already called out. However, most of them at this point are pretty obvious.

When we walk through the design of the wreck replacement, we should indeed nail these down if possible.

**Comment by chu11** [comment]
Most of the states are pretty clear once you read the "Program Interface" beneath.  I was initially confused by the "empty" state b/c I mentally mapped the states into those of a job queue.

**Comment by grondo** [comment]
Yeah, I apologize, I don't think we even use the `empty` state now in the wreck prototype. We have `reserved` (which is an entry that is created but essentially has no other state) and `submitted` which is a state that tells the scheduler the job wants to be scheduled.
